content/guides/digging-deeper/route-model-binding.md

```diff
@@ -0,0 +1,279 @@
+:::
+
+```ts
+// Routes file
+Route.get('posts/:post/comments/:comment', 'PostsController.show')
+```
+
+```ts
+// Controller
+import Post from 'App/Models/Post'
+import Comment from 'App/Models/Comment'
+import { bind } from '@adonisjs/route-model-binding'
+
+export default class PostsController {
+  @bind()
+  public async show({}, post: Post, comment: Comment) {
+    return { post, comment }
+  }
+}
+```
+
+
+:::note
+**Are you a visual learner**? Checkout [these screencasts](https://learn.adonisjs.com/series/route-model-binding/introduction) to learn about Route model binding, its setup and usage.
+:::
+
+
+## Basic usage
+Start with the most basic example and then tune up the complexity level to serve different use cases.
+
+In the following example, we will bind the `Post` model with the first parameter in the `posts/:id` route.
+
+```ts
+Route.get('/posts/:id', 'PostsController.show')
+```
+
+```ts
+import Post from 'App/Models/Post'
+import { bind } from '@adonisjs/route-model-binding'
+
+export default class PostsController {
+  @bind()
+  public async show({}, post: Post) {
+    return { post }
+  }
+}
+```
+
+:::warning
+Make sure you always import your model with an `import` and not an `import type`. Otherwise, the `bind` decorator will not be able to retrieve the class of your model and it will not work.
+
+**ESLint user?** - Take a look [here](#compatibility-with-eslint)
+:::
+
+The params and models are matched in the order they are defined. So the first param in the URL matches the first type-hinted model in the controller method.
+
+The match is not performed using the name of the controller method argument because TypeScript decorators cannot read them (so the technical limitation leaves us with the order-based matching only).
+
+## Changing the lookup key
+By default, the model's primary key is used to find a matching row in the database. You can either change that globally or change it for just one specific route.
+
+### Change lookup key globally via model
+After the following change, the post will be queried using the `slug` property and not the primary key. In a nutshell, the `Post.findByOrFail('slug', value)` query is executed.
+
+```ts
+class Post extends BaseModel {
+  public static routeLookupKey = 'slug'
+}
+```
+
+### Change the lookup key for a single route.
+In the following example, we define the lookup key directly on the route enclosed with parenthesis.
+
+```ts
+Route.get('/posts/:id(slug)', 'PostsController.show')
+```
+
+**Did you notice that our route now reads a bit funny?**\
+The param is written as `:id(slug)`, which does not translate well. Therefore, with Route model binding, we recommend using the model name as the route param because we are not dealing with the `id` anymore. We are fetching model instances from the database.
```

Comment:
```suggestion
The param is written as `:id(slug)`, which does not translate well. Therefore, with Route model binding, we recommend using the model name as the route param because we are no longer dealing with the `id`. Instead, we are fetching model instances from the database.
```